proposals/FileSystemSubscriptionManager.md

```diff
@@ -0,0 +1,125 @@
+
+Hence, simply enabling it doesn't satisfy the goal. This approach might only work for websites that either can ensure a long-running service worker or are not really interested in using service worker registration for observing the file change events after the websites are closed.
+
+[fso]: https://github.com/whatwg/fs/blob/main/proposals/FileSystemObserver.md#handling-changes-made-outside-the-lifetime-of-a-filesystemobserver
+[chrome-sw]: https://developer.chrome.com/blog/longer-esw-lifetimes#background
+
+#### Outliving Service Worker
+
+What if updating FileSystemObserver to allow it to outlive without the limit of running in ServiceWorkerGlobalScope?
+
+If such an option is implemented, there needs to be mechanisms to
+Tell when the browser should stop watching file changes.
+Handle file changes happen after the service worker is already terminated but the website is still open. Possibly needs a way to wake up a new service worker.
+
+There are existing mechanisms to auto wake up new service workers. Hence the next option.
+
+### Option 2. Utilizing Service Worker Registration
+
+The ServiceWorkerRegistration interface represents registration of a service worker for a specific origin and scope. The browser maintains a persistent list of active ServiceWorkerRegistration even when the associated service worker is not actively running, and will wake up new service workers if a registered event happens.
+
+## Proposed API
+
+### Web IDL
+
+First, define a new interface FileSystemSubscriptionManager that will be held under every ServiceWorkerRegistration, which manages subscriptions to changes to file systems:
+
+```idl
+partial interface ServiceWorkerRegistration {
+  // Returns a reference to FileSystemSubscriptionManager interface, which allows
+  // for subscribing to specific file changes.
+  [SameObject] readonly attribute FileSystemSubscriptionManager fileSystem;
+};
+```
+
+```idl
+// Provides methods for managing file system subscriptions.
+interface FileSystemSubscriptionManager {
+  // Subscribes to changes to a FileSystemHandle with the browser with specific
+  // options. Returns a Promise that resolves when the subscription completes.
+  Promise<void> subscribe(FileSystemHandle handle,
+                          FileSystemObserverObserverOptions options = {});
+  // Unsubscribes to changes to a FileSystemHandle. Returns a Promise that
+  // resolves when the unsubscription completes.
+  Promise<void> unsubscribe(FileSystemHandle handle);
+  // Returns a Promise that resolves with a list of FileSystemSubscription
+  // representing all the current file system subscriptions with the browser.
+  Promise<sequence<FileSystemSubscription>> getSubscriptions();
+};
+```
+
+```idl
+// Represents a subscription to changes to a FileSystemHandle.
+interface FileSystemSubscription {
```

Comment:
This does not have valid interface members. It looks more like a dictionary.

Reply:
Done.